### ISSUE TITLE
No bin folder under mantis/ITKStuff/Build.${ARCH}/MANTiS-build

### ISSUE THREAD

**Issue by lekai-crystal** [DevelopmentalImagingMCRI/mantis#24]
Hi, 
I have a question. After the installation of mantis, I found no "bin" folder under the "mantis/ITKStuff/Build.${ARCH}/MANTiS-build" directory. 
I use macos Ventura 13.2.1 with apple M2 chip, matlab2023b. 
Could you tell me how to solve the problem? Thanks a lot.
<img width="867" alt="截屏2024-05-12 21 37 24" src="https://github.com/DevelopmentalImagingMCRI/mantis/assets/54347996/083c1dea-5e9a-46d7-9492-25661ea518ed">

**Comment by lekai-crystal** [comment]
Though there is no "bin" folder, I can open the mantis GUI through spm GUI. Does it matter?
<img width="780" alt="截屏2024-05-12 21 47 09" src="https://github.com/DevelopmentalImagingMCRI/mantis/assets/54347996/2dd22c55-1c3a-4b8b-b64b-b37855a7ac2d">

**Comment by lekai-crystal** [comment]
I tired the mantis found it didn't work. 

I have read the previously raised issues and found the same question (No bin folder in MANTiS-build #13). 
I tired the solution:"Try deleting the cloned folder. Reclone, then do
cd mantis
git checkout BumpITKVersion"
but I failed.
The command showed:"error: pathspec 'BumpITKVersion' did not match any file(s) known to git"
I have installed git cmake and xcode.

Could you tell me how to solve the problem.

Thank you in advance.

Regards,
Lekai Luo

**Comment by richardbeare** [comment]
First up - I haven't tested on an M2 mac, hopefully we'll be able to do something.

The bin directory won't exist until you've run a cmake configuration step and build (which should take some time).

I can't see why the git checkout command didn't work. Are you sure you've run these commands:

```
git clone https://github.com/DevelopmentalImagingMCRI/mantis.git
cd mantis
git submodule init
git submodule update
```

before running

```
git checkout BumpITKVersion
```

**Comment by lekai-crystal** [comment]
Hi，
Thank you so much for your response. 
I deleted the mantis folder and reinstalled it as the way you suggested.
But it didn't work. There is still an error with "git checkout BumpITKVersion"
<img width="573" alt="截屏2024-05-13 21 36 39" src="https://github.com/DevelopmentalImagingMCRI/mantis/assets/54347996/ef5191e8-9cde-4935-a246-1714fc780c86">
I would really appreciate it if you could help me with this problem.

**Comment by lekai-crystal** [comment]
Then I tried the following steps:
```
cd ITKStuff
export ARCH=MACA64
mkdir Build.${ARCH} 
cd Build.${ARCH}
cmake ../SuperBuild
```
When running "cmake ../SuperBuild", there might be something wrong. I saw some steps failed as reported in the terminal window.
<img width="839" alt="截屏2024-05-13 22 40 53" src="https://github.com/DevelopmentalImagingMCRI/mantis/assets/54347996/47a84862-369a-402d-b801-4a03db434268">
There is no "bin" folder under the "MANTiS-build" directory. 
I'm looking forward to your reply.

Sincerely,
Lekai Luo

**Comment by richardbeare** [comment]
Ah - you need to run "make" after cmake. cmake is the configuration step, make does the building.

I think the BumpITKVersion branch must have been integrated into the main, then deleted.

**Comment by lekai-crystal** [comment]
Dear Richard, 
Thanks for your advice. After running "make" after "cmake", the bin folder under "MANTiS-build" is made. 

But there is still an error with "git checkout BumpITKVersion". As before, the terminal command window showed "error: pathspec 'BumpITKVersion' did not match any file(s) known to git".

Then I tried to use mantis through spm on matlab. The matlab command window showed:

> SPM: spm_mantis (1)                                22:46:34 - 14/05/2024
========================================================================
Item tools: No field(s) named
mantis
Item tools: No field(s) named
mantis
Item tools: No field(s) named
mantis
Item tools: No field(s) named
mantis
Item tools: No field(s) named
mantis
Item tools: No field(s) named
mantis
Item tools: No field(s) named
mantis
Item tools: No field(s) named
mantis
Item tools: No field(s) named
mantis
Item tools: No field(s) named
mantis
Item tools: No field(s) named
mantis
Item file_move: Dependency 'Mantis: Phase 1 tissue classification: Seg Params' could not be added.
Item file_move: Dependency 'Folders for phases - image files get placed here: Phase1 single subfolder' could not be added.
Item file_move: Dependency 'Mantis: Phase 1 tissue classification: c1 Images' could not be added.
Item file_move: Dependency 'Folders for phases - image files get placed here: Phase1 single subfolder' could not be added.
Item file_move: Dependency 'Mantis: Phase 1 tissue classification: c3 Images' could not be added.
Item file_move: Dependency 'Folders for phases - image files get placed here: Phase1 single subfolder' could not be added.
Item file_move: Dependency 'Mantis: Phase 1 tissue classification: c2 Images' could not be added.
Item file_move: Dependency 'Mantis: Phase 1 tissue classification: c4 Images' could not be added.
Item file_move: Dependency 'Mantis: Phase 1 tissue classification: c5 Images' could not be added.
Item file_move: Dependency 'Mantis: Phase 1 tissue classification: c6 Images' could not be added.
Item file_move: Dependency 'Mantis: Phase 1 tissue classification: c7 Images' could not be added.
Item file_move: Dependency 'Mantis: Phase 1 tissue classification: c8 Images' could not be added.
Item file_move: Dependency 'Mantis: Phase 1 tissue classification: c9 Images' could not be added.
Item file_move: Dependency 'Folders for phases - image files get placed here: Phase1 single subfolder' could not be added.
Item file_move: Dependency 'Mantis: Phase 1 tissue classification: Forward Deformations' could not be added.
Item file_move: Dependency 'Folders for phases - image files get placed here: Phase1 single subfolder' could not be added.
Item file_move: Dependency 'Mantis: Phase 1 tissue classification: Inverse Deformations' could not be added.
Item file_move: Dependency 'Folders for phases - image files get placed here: Phase1 single subfolder' could not be added.

I'm sorry to bother you again. But how to solve the problem? 

Sincerely,
Lekai Luo

**Comment by richardbeare** [comment]
At what point did those errors appear? Had you restarted matlab and spm after the mantis install process was complete?

I might need screenshots to solve this

**Comment by lekai-crystal** [comment]
Dear Richard,
    When I open mantis and click on mantis/mantis segmentation pipeline, the matlab command window will show the error.
<img width="591" alt="截屏2024-05-15 22 18 47" src="https://github.com/DevelopmentalImagingMCRI/mantis/assets/54347996/eb956516-14cb-4ba7-b290-e5a81ef0cffa">
<img width="1403" alt="截屏2024-05-15 22 28 29" src="https://github.com/DevelopmentalImagingMCRI/mantis/assets/54347996/b4b768ac-16b3-418b-9d4a-482ba3caf460">
<img width="1417" alt="截屏2024-05-15 22 28 53" src="https://github.com/DevelopmentalImagingMCRI/mantis/assets/54347996/7bebd8bc-2df1-49ee-9db2-3664e913bc59">
    I'm using the development version of spm, which support the macbook with apple silicon (https://github.com/spm/spm/). The common spm12 doesn't work on my macbook.
    So what is the problem? 
    
Sincerely,
Lekai

**Comment by richardbeare** [comment]
I'm afraid I have no insight into this. I'll do some tests on my M1 mac, but can't promise anything. It will take some time, I'm afraid.

**Comment by richardbeare** [comment]
I think we may be over the first hurdle - there is at least something for you to test. 

go to your mantis folder and run

```
git pull
git checkout DevSPMSupport
```

Then test again - no need to rebuild etc.

The problem I have found was caused by a change in a core spm function about 3 years ago, leading to a structure element being returned as a function handle instead of the results of a function call.

I've only tested that this fixes the first problem, which it seems to. There are likely to be others as you work your way through.
